ssl/tls13_enc.c

```diff
@@ -764,12 +772,132 @@ int tls13_change_cipher_state(SSL *s, int which)
         s->statem.enc_write_state = ENC_WRITE_STATE_WRITE_PLAIN_ALERTS;
     else
         s->statem.enc_write_state = ENC_WRITE_STATE_VALID;
+#ifndef OPENSSL_NO_KTLS
+# if defined(OPENSSL_KTLS_TLS13)
+#  ifndef __FreeBSD__
+    if (!(which & SSL3_CC_WRITE)
+        || ((which & SSL3_CC_WRITE) && (s->mode & SSL_MODE_NO_KTLS_TX)))
+        goto skip_ktls;
+
+    /* ktls supports only the maximum fragment size */
+    if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
+        goto skip_ktls;
+
+    /* check whether cipher is known */
+    if (cipher == NULL)
+        goto skip_ktls;
+
+    /* check that cipher is AES_GCM_128, AES_GCM_256, AES_CCM_128 */
+    switch (EVP_CIPHER_nid(cipher))
+    {
+#   ifdef OPENSSL_KTLS_AES_GCM_128
+    case NID_aes_128_gcm:
+        if (EVP_CIPHER_mode(cipher) != EVP_CIPH_GCM_MODE ||
+            EVP_CIPHER_key_length(cipher) != TLS_CIPHER_AES_GCM_128_KEY_SIZE)
```

Comment:
Again I'm not sure why we need more than just checking the nid.

Reply:
removed this check in `ktls_check_supported_cipher`